src/components/SharedComponents/LocationDisplay.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+
+import { Body4 } from "components/SharedComponents";
+import { View } from "components/styledComponents";
+import * as React from "react";
+import { useTranslation } from "react-i18next";
+import IconMaterial from "react-native-vector-icons/MaterialIcons";
+import useLocationName from "sharedHooks/useLocationName";
+
+type Props = {
+  latitude: number,
+  longitude: number
+};
+
+const LocationDisplay = ( {
+  latitude,
+  longitude
+}: Props ): React.Node => {
+  const { t } = useTranslation( );
+  const locationName = useLocationName( latitude, longitude );
```

Comment:
Sorry, I'm going to change the requirements here: this is for displaying observations, potentially a lot of them, so we don't want to potentially reverse geocode coordinates 100s of times if you're perusing your observations. Instead, this component should take an `observation` as a prop and do something like this pseudocode:

```
if placeGuess is not blank:
  show placeGuess
else if latitude and longitude are not blank
  show `${latitude}, ${longitude}`
else
  show "Missing Location"
```

Let me know if that's confusing at all.

Reply:
Makes sense to me.

will a missing longitude/latitude be `null`? Thinking about the one case where an observation is 0,0 (if there's even been one at that location)

Reply:
In theory a missing coordinate should be null, and 0,0 should be interpreted as literally at 0,0 (the only obs marooned on [Null Island](https://en.wikipedia.org/wiki/Null_Island) are due to bugs, but they do exist and should be treated as if they were really there).